packages/api/cms-api/src/file-uploads/file-uploads.controller.ts

```diff
@@ -0,0 +1,45 @@
+import { EntityManager } from "@mikro-orm/postgresql";
+import { Controller, Post, Type, UploadedFile, UseInterceptors } from "@nestjs/common";
+import rimraf from "rimraf";
+
+import { DisableCometGuards } from "../auth/decorators/disable-comet-guards.decorator";
+import { FileUploadInput } from "../dam/files/dto/file-upload.input";
+import { RequiredPermission } from "../user-permissions/decorators/required-permission.decorator";
+import { FileUpload } from "./entities/file-upload.entity";
+import { FileUploadsService } from "./file-uploads.service";
+import { FileUploadsFileInterceptor } from "./file-uploads-file.interceptor";
+
+export function createFileUploadsController(options: { public: boolean }): Type<unknown> {
+    @Controller("file-uploads")
+    class BaseFileUploadsController {
+        constructor(private readonly fileUploadsService: FileUploadsService, private readonly entityManager: EntityManager) {}
+
+        @Post("upload")
+        @UseInterceptors(FileUploadsFileInterceptor("file"))
+        async upload(@UploadedFile() file: FileUploadInput): Promise<FileUpload> {
+            const fileUpload = await this.fileUploadsService.upload(file);
+
+            await this.entityManager.flush();
+
+            rimraf(file.path, (error) => {
+                if (error) {
+                    console.error("An error occurred when removing the file: ", error);
+                }
+            });
+
+            return fileUpload;
+        }
+    }
+
+    if (options.public) {
+        @DisableCometGuards()
```

Comment:
I'm now setting the decorator on the controller, not the endpoint, to remove code duplication. Is this okay, or should I change it back?

Reply:
I'd say for now it's okay. But maybe we have to change this in the future. I'm thinking of a scenario where upload is private but download/view should be public.

Reply:
Download/view could be in another controller